google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java

```diff
@@ -237,16 +248,26 @@ private void showWarnings() {
     if (!isValidConfigurationFile(appYaml.getText())) {
       errorMessage.setText(GctBundle.getString("appengine.deployment.error.staging.yaml"));
       showError = true;
-    }
-
-    try {
-      if (isRuntimeCustom() && !isValidConfigurationFile(dockerfile.getText())) {
-        errorMessage.setText(GctBundle.getString("appengine.deployment.error.staging.dockerfile"));
+    } else {
+      try {
+        if (isRuntimeCustom()) {
+          String dockerDirectoryTest = dockerDirectory.getText();
+          if (dockerDirectoryTest.isEmpty() || !Files.isDirectory(Paths.get(dockerDirectoryTest))) {
```

Comment:
dockerDirectoryTe**x**t? or dockerDirectoryName? also you could do something like  StringUtils.isEmpty for a null safe check too.